src/openfermion/utils/_unitary_cc.py

```diff
@@ -117,17 +118,25 @@ def uccsd_singlet_paramsize(n_qubits, n_electrons):
         Number of independent parameters for singlet UCCSD with a single
         reference.
     """
-    n_occupied = int(numpy.ceil(n_electrons / 2.))
-    n_virtual = n_qubits // 2 - n_occupied
+    if n_electrons % 2 != 0:
```

Comment:
This is actually not strictly a requirement, a singlet operator can be applied to a doublet state for example.  However I'm not sure how thoroughly we've tested those cases, so perhaps it would be okay to change the error message to a warning instead saying something like "untested for odd number of electrons"?

Reply:
No, I realize now there should be no problem if the input is odd. I'll fix this.

Reply:
Sounds good.  I realize the existing code would probably need to be generalized to allow application to triplet states, though I'm not sure about doublet states.  Maybe we can leave the error in for now and change the message to "not implemented".  What are your thoughts?

Reply:
How should we handle an odd value of `n_qubits`? The old code seems inconsistent on this.

Reply:
It should at least be a not implemented error, as an odd number of qubits is incompatible with any encoding we've specified  I think.  That would have needed to result from a manual truncation of qubits, which if done, should have been done after calling these routines I would think.